Fix warnings in doc comments

ashawley opened the pull request:
- Fixes a few formatting problems
- Fixes a bunch of warnings about cross-reference links
- Reduces the number of warnings in half

Follow-up to #8284, #8290, and #8339.

ashawley commented:
If someone happens to want to spot check any of these changes, I've published everything here:

https://ashawley.github.io/scala/2.13.0/compiler/
https://ashawley.github.io/scala/2.13.0/library/
https://ashawley.github.io/scala/2.13.0/partest/
https://ashawley.github.io/scala/2.13.0/reflect/
https://ashawley.github.io/scala/2.13.0/scalap/
https://ashawley.github.io/scala/2.13.0/testkit/

SethTisue reviewed (approved):
wow, super!

som-snytt commented on src/compiler/scala/tools/nsc/ObjectRunner.scala:
```diff
@@ -21,9 +21,9 @@ trait CommonRunner {
-   *  @throws InvocationTargetException
+   *  @throws java.lang.ClassNotFoundException
+   *  @throws java.lang.NoSuchMethodException
+   *  @throws java.lang.reflect.InvocationTargetException
```
This gets an exceptions thrown clause but no exceptions. That is not your fault, I don't know how it works but was curious to pick something and see what it looked like. https://ashawley.github.io/scala/2.13.0/compiler/scala/tools/nsc/ObjectRunner$.html

ashawley replied:
Yeah, you found a good one.  I don't know what that is about...  Experimenting with it further, it would seem it's because there is no text description after the exception class name.  I fixed the warning about "unable to find link" by adding `java.lang`, but it seems we need a new warning in Scaladoc for when the description is missing.

ashawley replied:
Yup, that's the issue

![Screen Shot 2019-08-20 at 9 55 41 PM](https://user-images.githubusercontent.com/358615/63396521-52e82600-c395-11e9-9e83-02be698a3340.png)

som-snytt commented on src/compiler/scala/tools/nsc/typechecker/NamesDefaults.scala:
```diff
@@ -118,7 +118,7 @@ trait NamesDefaults { self: Analyzer =>
    *
    * @param tree: the function application tree
-   * @argPos: a function mapping arguments from their current position to the
+   * @param argPos: a function mapping arguments from their current position to the
```
Is it possible I'm looking at the before and not the after? https://ashawley.github.io/scala/2.13.0/compiler/scala/tools/nsc/typechecker/NamesDefaults.html#transformNamedApplication(typer:NamesDefaults.this.Typer,mode:scala.tools.nsc.Mode,pt:NamesDefaults.this.global.Type)(tree:NamesDefaults.this.global.Tree,argPos:Int=%3EInt):NamesDefaults.this.global.Tree

ashawley replied:
I'm seeing the same issue, locally.  I wonder if it's the blank line on line 119?

ashawley replied:
Actually, it's the colons...

som-snytt replied:
People should go in to have their colons checked per their doctor's recommendation.

ashawley replied:
I've fixed the colons.

ashawley commented on src/reflect/scala/reflect/api/Internals.scala:
```diff
@@ -194,7 +194,7 @@ trait Internals { self: Universe =>
 
     /** This symbol cast to a free type symbol.
-     *  @throws ScalaReflectionException if `isFreeType` is false.
+     *  @throws scala.reflect.ScalaReflectionException if `isFreeType` is false.
```
I'm wrong here.  It's actually `scala.ScalaReflectionException`.

som-snytt replied:
That feels like a bug, doesn't it? Also, Internals has nested package statements so it should be visible in any case and qualification in scaladoc should not be necessary. (Not that I know how it actually works.)

ashawley replied:
Good point.  It's not a bug in Scaladoc, but is caused by https://github.com/scala/bug/issues/11706, which is an sbt config issue, so Scaladoc can't resolve it since `ScalaReflectionException` is defined elsewhere in scala-library.  Adding `autoAPIMappings` does resolve correctly for the reasons you suggest.  I'll revert this, then.  The warnings will get fixed with 11706.

som-snytt replied:
Oh, I see. BTW I actually meant it looks like a bug or accident that the exception is not in scala.reflect.

ashawley replied:
That's funny.  I actually had that interpretation of your question, but concluded you were raising a package resolution problem about Scaladoc.

som-snytt replied:
Yes, I meant that too.

ashawley replied:
Ah, ok.  You really meant "Also".

ashawley committed:
Fix warnings in doc comments

Addressing these warnings fixes a lot of syntax issues that corrects a
lot of broken cross-referencing links, and it also cleans up some
formatting problems.

som-snytt commented:
I wonder if this is the same transient failure that @SethTisue noticed on travis:
```
(file:/home/jenkins/workspace/scala-2.13.x-validate-main/build/quick/classes/library/ <no signer certificates>)
java.nio.file.NoSuchFileException: /tmp/scala.picklecache1350699566304902807/tmp/pipelineBase2596684608616952144/projects/m1/p1/target.jar
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixPath.toRealPath(UnixPath.java:837)
	at scala.tools.nsc.PipelineMainClass.$anonfun$newCompiler$1(PipelineMain.scala:622)
	at scala.tools.nsc.PipelineMainClass.newCompiler(PipelineMain.scala:619)
	at scala.tools.nsc.PipelineMainClass$Task.liftedTree1$1(PipelineMain.scala:432)
	at scala.tools.nsc.PipelineMainClass$Task.compiler$lzycompute(PipelineMain.scala:431)
	at scala.tools.nsc.PipelineMainClass$Task.compiler(PipelineMain.scala:431)
	at scala.tools.nsc.PipelineMainClass.$anonfun$process$57(PipelineMain.scala:261)
	at scala.tools.nsc.PipelineMainClass.$anonfun$process$57$adapted(PipelineMain.scala:261)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:447)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

SethTisue commented:
indeed, attn Señor Picklecachemeister Zaugg @retronym

SethTisue commented:
/rebuild
